src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisChatSessionResource.java

```diff
@@ -38,8 +35,9 @@ public class IrisChatSessionResource extends IrisExerciseChatBasedSessionResourc
 
     protected IrisChatSessionResource(AuthorizationCheckService authCheckService, IrisChatSessionRepository irisChatSessionRepository, UserRepository userRepository,
             ProgrammingExerciseRepository programmingExerciseRepository, IrisSessionService irisSessionService, IrisSettingsService irisSettingsService,
-            IrisRateLimitService irisRateLimitService) {
-        super(authCheckService, userRepository, irisSessionService, irisSettingsService, irisRateLimitService, programmingExerciseRepository::findByIdElseThrow);
+            PyrisHealthIndicator pyrisHealthIndicator, IrisRateLimitService irisRateLimitService) {
+        super(authCheckService, userRepository, irisSessionService, irisSettingsService, pyrisHealthIndicator, irisRateLimitService,
+                programmingExerciseRepository::findByIdElseThrow);
```

Comment:
avoid repository calls in the constructor!

Reply:
This is actually not a repository call, but a reference to a repository method, which is then store in `protected final Function<Long, E> exerciseByIdFunction` in the super class. 
It would be useful to do it like this if we were to support different exercise types here, but this is not really planned right now, so I can also simplify this.

Reply:
Changed it.